### PR TITLE
feat: highlight active footer navigation

### LINKED ISSUE #58: 🚀 Feature: Highlight Active Footer Navigation
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/philly-js-club/philly-js-club-website-remix/issues?q=is%3Aissue) and found none that matched my issue.

## User story

As a user, I want the site navigation to communicate where I am in the site.

## Acceptance criteria

- Links in the site footer should the color white when the user is on that page
    - As of creating this issue this applies to:
        - About
        - Code of Conduct
        - Events
        - Sponsors
        - Join Us

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by RNR1**
## PR Checklist

- [x] Addresses an existing open issue: fixes #058
- [x] That issue was marked as [`status: accepting prs`](https://github.com/philly-js-club/philly-js-club-website-remix/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/philly-js-club/philly-js-club-website-remix/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

Submitting it again, using `NavLink` with the `reloadDocument` prop. It works like the native anchor tag but will give us the "active" state out of the box.

### CODE REVIEW

File: app/shared.css [hunk 1]
```diff
@@ -151,6 +151,10 @@ Let us know if you think you've found one we'd like. 😏
 	padding: 0;
 }
 
+.page-grid-footer-link.active {
+	color: white;
```
Comment (JoshuaKGoldberg) [inline app/shared.css]: A couple of issues here:

* Other colors use [CSS variables](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties) to also support [dark mode](https://developer.chrome.com/docs/devtools/rendering/emulate-css)
* White on our yellow has [not enough color contrast to be accessible](https://webaim.org/resources/contrastchecker/?fcolor=FFFFFF&bcolor=EFDB4F)
    * If you hover your mouse over the element while the Chrome dev tools > Elements tab is open, it'll give you a nice display including the ratio:
    Chrome dev tools elements hover showing a 1.4 contrast ratio

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Thanks for sending this! A good start - but let's tweak the visual display to be accessible and support dark mode.

Sorry for the wait btw - I'd forgotten to subscribe to this repo's events 🤦.

File: app/shared.css [hunk 2]
```diff
@@ -151,6 +151,10 @@ Let us know if you think you've found one we'd like. 😏
 	padding: 0;
 }
 
+.page-grid-footer-link.active {
+	text-decoration: underline;
```
Comment (JoshuaKGoldberg) [inline app/shared.css:155]: This works! I do want to think a bit more holistically on the design system (we're starting to have a bunch of different treatments for emphasis)... but I like this a lot for now. 😄 

cc @codylewis @tjwds - thoughts?

**Review by JoshuaKGoldberg — APPROVED** [review]
Looks good to me!

I'll wait a bit to see if anybody else wants to chime in 😄 but I feel happy about this. Thanks! 🙌

File: app/shared.css (hunk 2 above)
Comment (codylewis) [reply app/shared.css:155]: Agreed on the holistic design system comment and that this work for now 👌

Comment (tjwds) [reply app/shared.css:155]: Yep, agreed, thanks!  I filed #69.

Comment (JoshuaKGoldberg) [reply app/shared.css:155]: Whoo!

**Pull request merged**